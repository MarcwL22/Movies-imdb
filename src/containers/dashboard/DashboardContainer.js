import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchMovies } from './duck';
import { fetchMovie } from '../movie/duck';
// UIComponents
import { Container, Heading2 } from '../../components';
// Components
import MovieContainer from '../movie/MovieContainer';
import DashboardHeader from './components/DashboardHeader';
import DashboardGrid from './components/DashboardGrid';
import styled from 'styled-components';
// Assets
import { BREAKPOINTS } from '../../assets/variables';

const GridTitleContainer = styled(Container)`
  padding: 0 3.2rem;
  @media (max-width: ${BREAKPOINTS.phone}) {
    padding: 0 1.6rem;
  }
`;

export class DashboardContainer extends Component {
  state = { searchText: '' };

  componentDidMount = () => {
    this.props.fetchMovies('Avengers');
  };

  formSubmit = event => {
    console.log(this.state.searchText);
    event.preventDefault();
    this.props.fetchMovies(this.state.searchText);
  };

  onChangeText = event => {
    this.setState({ searchText: event.target.value });
    this.props.fetchMovies(this.state.searchText);
  };

  render() {
    const { searchText } = this.state;
    const { movies, loading, error } = this.props;
    return (
      <div>
        <DashboardHeader onSubmit={this.formSubmit} value={searchText} onChange={this.onChangeText} />
        <GridTitleContainer>
          <Heading2>{searchText}</Heading2>
        </GridTitleContainer>
        <DashboardGrid loading={loading} error={error} data={movies} onPress={this.props.fetchMovie} />
        <MovieContainer />
      </div>
    );
  }
}

const mapStateToProps = ({ dashboardReducer }) => {
  return { ...dashboardReducer };
};

export default connect(
  mapStateToProps,
  { fetchMovies, fetchMovie }
)(DashboardContainer);
