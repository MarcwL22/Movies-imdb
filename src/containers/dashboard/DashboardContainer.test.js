import React from 'react';
import { shallow, mount } from 'enzyme';
import { DashboardContainer } from './DashboardContainer';

describe('DashboardContainer', () => {
  let props = {
    movies: {},
    loading: false,
    error: false
  };
  let dashboard = shallow(<DashboardContainer {...props} />);
  it('should renders properly', () => {
    expect(dashboard).toMatchSnapshot();
  });
  describe.skip('should when mounted', () => {
    const mockFetchMovies = jest.fn();
    beforeEach(() => {
      props.fetchMovies = mockFetchMovies;
      dashboard = mount(<DashboardContainer {...props} />);
    });
    it('should dispatches the `fetchMovies()` method it receives from props', () => {
      expect(mockFetchMovies).toHaveBeenCalled();
    });
  });
});
