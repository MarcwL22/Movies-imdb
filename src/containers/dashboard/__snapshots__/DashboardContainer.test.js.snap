// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardContainer should renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DashboardContainer
    error={false}
    loading={false}
    movies={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <DashboardHeader
          onChange={[Function]}
          onSubmit={[Function]}
          value="Avengers"
/>,
        <Styled(styled.div)>
          <styled.h2>
                    Avengers
          </styled.h2>
</Styled(styled.div)>,
        <Unknown
          data={Object {}}
          error={false}
          loading={false}
          onPress={[Function]}
/>,
        <Connect(MovieContainer) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onChange": [Function],
          "onSubmit": [Function],
          "value": "Avengers",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.h2>
            Avengers
</styled.h2>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Avengers",
          },
          "ref": null,
          "rendered": "Avengers",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Object {},
          "error": false,
          "loading": false,
          "onPress": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <DashboardHeader
            onChange={[Function]}
            onSubmit={[Function]}
            value="Avengers"
/>,
          <Styled(styled.div)>
            <styled.h2>
                        Avengers
            </styled.h2>
</Styled(styled.div)>,
          <Unknown
            data={Object {}}
            error={false}
            loading={false}
            onPress={[Function]}
/>,
          <Connect(MovieContainer) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onChange": [Function],
            "onSubmit": [Function],
            "value": "Avengers",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.h2>
              Avengers
</styled.h2>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Avengers",
            },
            "ref": null,
            "rendered": "Avengers",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Object {},
            "error": false,
            "loading": false,
            "onPress": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
